spec.go

```diff
@@ -248,7 +248,8 @@ func (d *Document) ResetDefinitions() *Document {
 
 // Pristine creates a new pristine document instance based on the input data
 func (d *Document) Pristine() *Document {
-	dd, _ := Analyzed(d.Raw(), d.Version())
```

Comment:
What does the member Raw() and the re-marshalled raw differ, or is this needed just for breaking circularity?

Reply:
The point with this PR is to capture the "new" raw whenever the content of th spec pointer is modified (you know everything is by pointers and a lot of this is mutable, hence a lot of our woes...).

When you create a new doc, the existing is just fine.
Now if you alter the spec (e.g. by flattening it), the Raw() is just pointless as it captures the original spec (which is saved elsewhere btw).

I'd like to use "Pristine" as a convenient way to start fresh and workaround this issue that I've documented there: https://github.com/go-openapi/spec/issues/180, that I could not solve easily (circular $ref are such a headache) and that is blocking me for solving this: https://github.com/go-swagger/go-swagger/pull/3020

Reply:
In go-swagger/generator/spec.go, there are a few initializations that alter the spec.
It is mostly ok, but in this special case of circular $ref expansion...